src/components/20-molecules/datepicker/index.js

```diff
@@ -0,0 +1,369 @@
+import datepickerCSS from './index.scss';
+import { getWeekdays, getAllLocaleMonthsArray, parseLocalisedDateIfValid } from './utils/date';
+import Store from './utils/Store';
+
+// TODO: Replace with icon in separate PR
+const iconDatepicker = svg`<svg class="m-datepicker__input-svg" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+<path fill="none" d="M0 0h24v24H0z"/>
+<path d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"/>
+</svg>
+`;
+export class Datepicker extends LitElement {
+  static tagName = 'axa-datepicker';
+  static styles = css`
+    ${unsafeCSS(datepickerCSS)}
+  `;
+
+  static get properties() {
+    return {
+      open: { type: Boolean, reflect: true },
+      locale: { type: String, reflect: true },
+      date: { type: Object, reflect: true },
+      outputDate: { type: String, reflect: true },
+      year: { type: Number, reflect: true },
+      month: { type: Number, reflect: true },
+      day: { type: Number, reflect: true },
+      inverted: { type: Boolean, reflect: true },
+      inputField: { type: Boolean },
+      monthItems: { type: Array },
+      yearItems: { type: Array },
+      cells: { type: Array },
+      allowedYears: { type: Array },
+      labelButtonCancel: { type: String },
+      labelButtonOk: { type: String },
+    };
+  }
+
+  // TODO: set the selectd item of the year and month array from startdate
+  constructor() {
+    super();
+    this.locale = 'de-ch'; // TODO get from attributes
+    this.labelButtonCancel = 'Cancel'; // TODO get from attributes
+    this.labelButtonOk = 'OK'; // TODO get from attributes
+    this.startDate = new Date(); // TODO get from attributes
+    this.date = this.startDate;
+    this.outputDate = '';
+    this.year = this.date.getFullYear();
+    this.month = this.date.getMonth();
+    this.allowedYears = [2019, 2020]; // TODO get from attributes
+    this.open = false;
+
+    this.monthItems = getAllLocaleMonthsArray(this.locale).map((item, index) => ({
+      isSelected: index === this.month,
+      name: item.toString(),
+      value: index.toString(),
+    }));
+
+    this.yearItems = this.allowedYears.map(item => ({
+      isSelected: item === this.year,
+      name: item.toString(),
+      value: item.toString(),
+    }));
+
+    this.store = new Store(this.locale, this.startDate, this.allowedYears);
+    this.cells = this.store.getCells();
+    this.weekdays = getWeekdays(this.startDate, this.locale);
+  }
+
+  debounce(func, wait, immediate) {
+    let timeout;
+    return (...args) => {
+      const context = this;
+      const later = () => {
+        timeout = null;
+        if (!immediate) func.apply(context, args);
+      };
+      const callNow = immediate && !timeout;
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+      if (callNow) func.apply(context, args);
```

Comment:
i think this is not hard defined anywere, but i personally see this kind of writing as dangerous. I would suggest to use this instead:

```
if (callNow) {
    func.apply(context, args);
}
```

I have seen so many errors evolving from this. But cause is not defined and per CoC you dont have to change it :)

Reply:
same for line 75

Reply:
This is a copy paste from a gist. so feel free to adapt.